### ISSUE TITLE
Firestore: Support returning the commit time from RunTransaction

### ISSUE THREAD

**Issue by galenwarren** [googleapis/google-cloud-go#6841]
**Is your feature request related to a problem? Please describe.**
When a transaction is performed in Firestore via ```RunTransaction```, the underlying commit operation returns a ```CommitResponse```, which contains the "commit time", as described [here](https://firebase.google.com/docs/firestore/transaction-data-contention). This response - including the commit time -- is ignored in the ```RunTransaction``` code,  [here](https://github.com/googleapis/google-cloud-go/blob/1a2064d8591a56025dd8a72564449cc853d7ca68/firestore/transaction.go#L142).

I would like to have access to the commit time. As described in the link above:

> Cloud Firestore guarantees serializable isolation of transactions. Transactions in Cloud Firestore are serialized and isolated by commit time.

My particular use case involves building and maintaining a data cache. Similar timestamps are available already ```DocumentSnapshot``` and ```QuerySnapshot```.

**Describe the solution you'd like**
Add a new ```TransactionOption``` -- ```GetCommitTime``` -- that would allow the caller to provide the address of a  ```time.Time``` where the commit time should be recorded upon successful commit.

**Describe alternatives you've considered**
If a new ```TransactionOption``` isn't used, the only other way to get the commit time would be to change the signature of ```RunTransaction``` or add a new method (```RunTransactionWithCommitTime```?) with a different signature. Those options seem messier, to me.

**Additional context**
I have this implemented in a fork and would be happy to contribute the code. Thanks.

**Comment by galenwarren** [comment]
@telpirion Would you be interested in a PR for this?

**Comment by telpirion** [comment]
@galenwarren Yes, please! If you're willing to contribute a PR, I'll gladly review it.

**Comment by galenwarren** [comment]
@telpirion Great, I'll pull something together here soon. Thanks.

**Comment by galenwarren** [comment]
@telpirion Here you go: https://github.com/googleapis/google-cloud-go/pull/6967. Thanks!

**Comment by bhshkh** [comment]
Fixed in https://github.com/googleapis/google-cloud-go/pull/6967

Released in 1.18.0
